code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm

```diff
@@ -359,4 +359,407 @@ Difficulty: Very Hard
+			NewTerrainFloors = /turf/open/floor/plating/abductor
+			NewTerrainWalls = /turf/closed/wall/mineral/abductor
+			NewTerrainChairs = /obj/structure/bed/abductor //ayys apparently don't have chairs. An entire species of people who only recline.
+			NewTerrainTables = /obj/structure/table/abductor
+
+/obj/machinery/anomalous_crystal/theme_warp/ActivationReaction(mob/user, method)
+	if(..())
+		var/area/A = get_area(src)
+		if(!A.outdoors && !(A in affected_targets))
+			for(var/atom/Stuff in A)
+				if(isturf(Stuff))
+					var/turf/T = Stuff
+					if((istype(T, /turf/open/space) || istype(T, /turf/open/floor)) && NewTerrainFloors)
+						var/turf/open/O = T.ChangeTurf(NewTerrainFloors)
+						if(O.air)
+							var/datum/gas_mixture/G = O.air
+							G.copy_from_turf(O)
+						if(FloorLayerChange)
+							O.layer = FloorLayerChange
+						if(prob(florachance) && NewFlora.len)
+							var/density_checks = 1
+							for(var/atom/junk in O)
+								if(junk.density)
+									density_checks = 0
+									break
+							if(density_checks)
+								var/atom/Picked = pick(NewFlora)
+								new Picked(O)
+						continue
+					if(istype(T, /turf/closed/wall) && NewTerrainWalls)
+						T.ChangeTurf(NewTerrainWalls)
+						continue
+				if(istype(Stuff, /obj/structure/chair) && NewTerrainChairs)
+					var/obj/structure/chair/Original = Stuff
+					var/obj/structure/chair/C = new NewTerrainChairs(Original.loc)
+					C.dir = Original.dir
+					qdel(Stuff)
+					continue
+				if(istype(Stuff, /obj/structure/table) && NewTerrainTables)
+					new NewTerrainTables(Stuff.loc)
+					qdel(Stuff)
+					continue
+			affected_targets += A
+
+/obj/machinery/anomalous_crystal/emitter //Generates a projectile when interacted with
+	activation_method = "touch"
+	cooldown_add = 50
+	var/generated_projectile = /obj/item/projectile/beam/emitter
+
+/obj/machinery/anomalous_crystal/emitter/New()
+	..()
+	generated_projectile = pick(/obj/item/projectile/beam/emitter,/obj/item/projectile/magic/fireball/infernal,/obj/item/projectile/magic/spellblade,
+								/obj/item/projectile/energy/net, /obj/item/projectile/bullet/meteorshot, /obj/item/projectile/beam/xray)
+
+/obj/machinery/anomalous_crystal/emitter/ActivationReaction(mob/user, method)
+	if(..())
+		var/obj/item/projectile/P = new generated_projectile(get_turf(src))
+		P.dir = dir
+		switch(dir)
+			if(NORTH)
+				P.yo = 20
+				P.xo = 0
+			if(EAST)
+				P.yo = 0
+				P.xo = 20
+			if(WEST)
+				P.yo = 0
+				P.xo = -20
+			else
+				P.yo = -20
+				P.xo = 0
+		P.fire()
+
+/obj/machinery/anomalous_crystal/dark_reprise //Revives anyone nearby, but turns them into shadowpeople. Cannot revive shadowpeople, so this is a one time thing.
+	activation_method = "touch"
+
+/obj/machinery/anomalous_crystal/dark_reprise/ActivationReaction(mob/user, method)
+	if(..())
+		for(var/i in range(1, src))
+			if(ishuman(i))
+				var/mob/living/carbon/human/H = i
+				if(H.stat == DEAD && !is_species(H, /datum/species/shadow))
```

Comment:
ree why isn't there an isshadow() define THIS ISN'T YOUR FAULT BUT FIX IT ANYWAY